### ISSUE TITLE
MacOS raw mouse input

### ISSUE THREAD

**Issue by expikr** [libsdl-org/SDL#10146]
I’m completely unfamiliar with MacOS and SDL’s support goals for it, but more recent versions of MacOS has [GCMouseInput](https://developer.apple.com/documentation/gamecontroller/gcmouseinput) that provides raw deltas instead of accelerated motion. Does SDL use this? From what I could see it still handles regular cocoa events, no?

I think by default MacOS should use the GCMouseInput API, and only handle accelerated delta if SDL_MOUSE_RELATIVE_SYSTEM_SCALE is enabled. 

And tangentially MacOS implementations should also be aware of the SDL_MOUSE_RELATIVE_MODE_CENTER hint.

**Comment by slouken** [comment]
Yes, this all makes sense. CGMouse has been hooked up for iOS and could be hooked up similarly for macOS. It probably needs refactoring and support for the hints as you mentioned.

**Comment by dsvensson** [comment]
This would be very valuable for all FPS type games that currently have to resort to nasty hacks like discussed in this thread,

https://discourse.libsdl.org/t/osx-raw-input-still-gives-acceleration-proposed-solution/20359

**Comment by icculus** [comment]
This has been asked for since #67, believe it or not.  :)

**Comment by dsvensson** [comment]
Question is what it takes to make it work when sandboxed.

**Comment by icculus** [comment]
I think CGMouseInput works in the sandbox; the ancient solution was trying to read from HID devices directly, which is a giant no-no for security.

**Comment by dsvensson** [comment]
Yes, it's supposed to be the golden way forward. I'm building with a cmake generated Xcode project so I must be missing something. Whatever it is, the solution will be valuable for this issue as well, as others will try to sandbox the future SDL implementation. A simple Xcode generated project in Swift that implements a simple handler gets the events when sandboxed so it should work.

**Comment by ej-sanmartin** [comment]
This should be pretty straightforward to implement, as @slouken said there's an implementation using `CGMouse` in iOS (in `src/video/uikit/SDL_uikitevents.m`) that we can follow. `GCMouse` only works for macOS 11.00 or later, so would we want to have fallback to the current `NSEvent` based implementation for macOS? i think we should support to retain backwards compatability with older Mac devices

I wouldn't mind implementing and sending a PR for this soon. lmk wdyt

**Comment by slouken** [comment]
> I wouldn't mind implementing and sending a PR for this soon. lmk wdyt

Yes please!
